test/smoke/dashCore.js

```diff
@@ -1,18 +1,115 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
-
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  let nodeNames = networkConfig.inventory.masternodes.hosts;
+  nodeNames = nodeNames.concat(networkConfig.inventory['wallet-nodes'].hosts);
+  // TODO miner rpc call stuck. why?
+  // nodeNames = nodeNames.concat(networkConfig.inventory['miners'].hosts);
+  nodeNames = nodeNames.concat(networkConfig.inventory['full-nodes'].hosts);
+  nodeNames.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const networkInfo = await rpc.getNetworkInfo();
+        expect(networkInfo.error).to.be.equal(null);
+        expect(networkInfo.result.networkactive).to.be.equal(true);
+        expect(networkInfo.result.subversion).to.have.string(`(${networkConfig.network.type}=${networkConfig.network.name})/`);
+      });
+    });
   });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
+
+  // The number of blocks should be almost the same (-3/+3) and block hash of particular
+  // block height should be the same. Using `blocks`
+  // and `bestblockhash` from `GetBlockChainInfo`.
+  // Verify it on masternodes, wallet-nodes, full-nodes, miners
+  it('should have correct blockhash and blocks count', async () => {
+    const blockCounts = [];
+    const blockhashes = {};
+    let blockchainInfo;
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      const blockCount = await rpc.getBlockCount();
```

Comment:
Why do you get blocks height from `getBlockCount` if we already have it in `getBlockchainInfo` ?

Reply:
use only getBlockchainInfo now